File: utils/src/basic_operations.py
import os

import utils

def run(fd):
    ntests = 3
    nincr_small = 1000
    nincr_big = 10000000

    utils.test_sleep(fd, ntests)
    utils.test_for_sleep(fd, ntests)

    utils.test_add(fd, ntests, nincr_small)
    utils.test_add(fd, ntests, nincr_big)

    utils.test_add_if(fd, ntests, nincr_small)
    utils.test_add_if(fd, ntests, nincr_big)

    utils.test_subtract(fd, ntests, nincr_small)
    utils.test_subtract(fd, ntests, nincr_big)

    utils.test_subtract_if(fd, ntests, nincr_small)
    utils.test_subtract_if(fd, ntests, nincr_big)

    utils.test_multiply(fd, ntests, nincr_small)
    utils.test_multiply(fd, ntests, nincr_big)

    utils.test_multiply_if(fd, ntests, nincr_small)
    utils.test_multiply_if(fd, ntests, nincr_big)

    utils.test_divide(fd, ntests, nincr_small)
    utils.test_divide(fd, ntests, nincr_big)

    utils.test_divide_if(fd, ntests, nincr_small)
    utils.test_divide_if(fd, ntests, nincr_big)


if __name__ == "__main__":
    file_name = '/results/basic_operations/python_basic_operations.txt'
    path = utils.chop_suffix_from_path(os.path.dirname(os.path.abspath(__file__)), '/utils/src') + file_name
    with open(path, 'w') as fd:
        run(fd)

File: utils/src/basic_operations.jl
include("utils.jl")

function run(fd)
    ntests = 3
    nincr_small = 1000
    nincr_big = 10000000

    test_sleep(fd, ntests)
    test_for_sleep(fd, ntests)

    test_add(fd, ntests, nincr_small)
    test_add(fd, ntests, nincr_big)

    test_add_if(fd, ntests, nincr_small)
    test_add_if(fd, ntests, nincr_big)

    test_subtract(fd, ntests, nincr_small)
    test_subtract(fd, ntests, nincr_big)

    test_subtract_if(fd, ntests, nincr_small)
    test_subtract_if(fd, ntests, nincr_big)

    test_multiply(fd, ntests, nincr_small)
    test_multiply(fd, ntests, nincr_big)

    test_multiply_if(fd, ntests, nincr_small)
    test_multiply_if(fd, ntests, nincr_big)

    test_divide(fd, ntests, nincr_small)
    test_divide(fd, ntests, nincr_big)

    test_divide_if(fd, ntests, nincr_small)
    test_divide_if(fd, ntests, nincr_big)
end


file_name = "/results/basic_operations/julia_basic_operations.txt"
path = string(chop_suffix_from_path(pwd(), "/utils/src"), file_name)
open(path, "w") do fd
    run(fd)
end